File: primeFactorization.py
from math import sqrt

def primeFactorization(num):
    factors = []

    for i in range(2, int(sqrt(num) + 1)):

        if(num % i == 0):
            count = 0

            while(num % i == 0):
                num = num / i
                count +=1

            factors.append((i, count))

    if (num != 1):
        factors.append((num, 1))

    return factors

print(primeFactorization(10000000000000000))

File: primeFactorization.js
/*
    Find the prime factors of a given number
    return the prime factors and its frequency

    note: if a number does not have any factor till sqrt(n) then it is a prime number
*/

const primeFactorization = (n) => {
    const factors = [];
    for (let i = 2; i <= Math.sqrt(n); i++) {

        if(n % i === 0) {
            let count = 0
            while (n % i === 0) {
                n = n / i;
                count++
            }
            factors.push({
                factor: i,
                count
            });
        }

    }

    // at this point n = 1 or it is a prime number
    if (n !== 1) {
        factors.push({
            factor: n,
            count: 1
        })
    }
    return factors;
};

console.log(JSON.stringify(primeFactorization(2244555533333333333333333354), undefined, 2))

/*
    For primeFactorization(36)
    [
        {
            "factor": 2,
            "count": 2
        },
        {
            "factor": 3,
            "count": 2
        }
    ]

    = 2 x 2 x 3 x 3
*/